schema.ts

```diff
@@ -1615,75 +1645,90 @@ export function normalizeSchema<T extends Schema>(schema: T): T {
-    const normalizedSchema = {
-      attribution,
-      options,
-      codaType,
-      description,
-      displayProperty: displayProperty ? normalizeSchemaKey(displayProperty) : undefined,
-      featured: featured ? featured.map(normalizeSchemaKey) : undefined,
-      featuredProperties: featuredProperties ? featuredProperties.map(normalizeSchemaKey) : undefined,
-      fixedId,
-      id: id ? normalizeSchemaKey(id) : undefined,
-      identity,
-      idProperty: idProperty ? normalizeSchemaKey(idProperty) : undefined,
-      imageProperty: imageProperty ? normalizeSchemaPropertyIdentifier(imageProperty, normalized) : undefined,
-      includeUnknownProperties,
-      linkProperty: linkProperty ? normalizeSchemaPropertyIdentifier(linkProperty, normalized) : undefined,
-      mutable,
-      primary: primary ? normalizeSchemaKey(primary) : undefined,
-      properties: normalized,
-      snippetProperty: snippetProperty ? normalizeSchemaPropertyIdentifier(snippetProperty, normalized) : undefined,
-      subtitleProperties: subtitleProperties
-        ? subtitleProperties.map(subProp => normalizeSchemaPropertyIdentifier(subProp, normalized))
-        : undefined,
-      titleProperty: titleProperty ? normalizeSchemaPropertyIdentifier(titleProperty, normalized) : undefined,
-      type: ValueType.Object,
-    } as T;
-
-    return normalizedSchema;
+    // The `as T` here seems like a typescript bug... shouldn't the above typeguard be
+    // sufficient to define T === GenericObjectSchema?
+    return normalizeObjectSchema(schema) as T;
   }
   return schema;
 }
 
+export function normalizeObjectSchema(schema: GenericObjectSchema): GenericObjectSchema {
+  const normalizedProperties: ObjectSchemaProperties = {};
+  const {
+    attribution,
+    options,
+    codaType,
+    description,
+    displayProperty,
+    featured,
+    featuredProperties,
+    id,
+    identity,
+    idProperty,
+    imageProperty,
+    includeUnknownProperties,
+    linkProperty,
+    primary,
+    properties,
+    snippetProperty,
+    subtitleProperties,
+    titleProperty,
+    type,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    __packId,
+    ...rest
+  } = schema;
+  // Have TS ensure we don't forget about new fields in this function.
+  ensureNever<keyof typeof rest>();
+  for (const key of Object.keys(properties)) {
+    const normalizedKey = normalizeSchemaKey(key);
+    const props = properties[key];
+    const {fixedId, fromKey, mutable, originalKey, required} = props;
+    if (originalKey) {
+      throw new Error('Original key is only for internal use.');
+    }
+    const normalizedProps: ObjectSchemaProperty = {
+      fromKey: fromKey || (normalizedKey !== key ? key : undefined),
+      originalKey: key,
+    };
+    if (fixedId) {
+      normalizedProps.fixedId = fixedId;
+    }
+    if (mutable) {
+      normalizedProps.mutable = mutable;
+    }
+    if (required) {
+      normalizedProps.required = required;
+    }
```

Comment:
Any reason not to just pass these through as-is like before? In particular, for the boolean ones, this will technically be a change in output, because values specified as `false` will now be omitted. I would vote for not churning more than necessary here.

Reply:
Ah, definitely need to at least pass through `false`. I just reverted this churn and the other.